capsules/extra/src/dfrobot_rainfall_sensor.rs

```diff
@@ -0,0 +1,208 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2022.
+
+//! DFRobot Gravity Rainfall sensor using the I2C bus.
+//!
+//! <https://wiki.dfrobot.com/SKU_SEN0575_Gravity_Rainfall_Sensor>
+//! <https://github.com/DFRobot/DFRobot_RainfallSensor/tree/master>
+//!
+
+use core::cell::Cell;
+use kernel::hil::i2c::{self, I2CClient, I2CDevice};
+use kernel::hil::sensors::{RainFallClient, RainFallDriver};
+use kernel::hil::time::{Alarm, AlarmClient, ConvertTicks};
+use kernel::utilities::cells::{OptionalCell, TakeCell};
+use kernel::ErrorCode;
+
+pub const BUFFER_SIZE: usize = 4;
+
+const PID_REGISTER: u8 = 0x00;
+const TIME_RAINFALL_REGISTER: u8 = 0x0C;
+const RAIN_HOUR_REGISTER: u8 = 0x26;
+
+#[derive(Clone, Copy, PartialEq)]
+enum DeviceState {
+    Identify,
+    Normal,
+    StartRainFall(usize),
+    ContinueRainFall,
+    FinalRainFall,
+    Broken,
+}
+
+#[allow(dead_code)]
```

Comment:
Is this still needed? Seems like it is used

Reply:
Good catch, removed